packages/adblocker/src/engine/engine.ts

```diff
@@ -489,11 +542,31 @@ export default class FilterEngine extends EventEmitter<
   public update({
     newNetworkFilters = [],
     newCosmeticFilters = [],
+    newPreprocessors = [],
     removedCosmeticFilters = [],
     removedNetworkFilters = [],
+    removedPreprocessors = [],
   }: Partial<IListDiff>): boolean {
     let updated: boolean = false;
 
+    // Update preprocessors
+    if (newPreprocessors.length !== 0 || removedPreprocessors.length !== 0) {
+      updated = true;
+      const preservedFilters = this.preprocessors.update({
+        added: newPreprocessors,
+        removed: removedPreprocessors,
+      }).preservedFilters;
+
+      if (removedCosmeticFilters) {
+        removedCosmeticFilters = removedCosmeticFilters.filter(
+          (one) => !preservedFilters.has(one),
+        );
+      }
+      if (removedNetworkFilters) {
+        removedNetworkFilters = removedNetworkFilters.filter((one) => !preservedFilters.has(one));
+      }
+    }
```

Comment:
It's unclear if removed/added filters should be specified only as part of the added/removed preprocessors or also as part of the added/removed networkFilters/cosmeticFilters.

I think before implementing this part, it is needed to figure out the changes required in `generateDiff` function (from `lists.ts`) which is in charge of generating these diffs in the first place.

Reply:
I think it's ok to preserve this part as I see a use-case of direct calling `FiltersEngine.updateFromDiff`. This method is often called with a manual update of filters.

---

However, I should look at the `generateDiff` and `mergeDiff` function before deciding something about this.